src/Microsoft.Diagnostics.Runtime/src/Utilities/Symbols/DefaultSymbolLocator.Async.cs

```diff
@@ -125,18 +125,17 @@ private async Task<string> SearchSymbolServerForFile(string fileSimpleName, stri
                 }
             }
 
-            string result = await GetFirstNonNullResult(tasks).ConfigureAwait(false);
+            string? result = await GetFirstNonNullResult(tasks).ConfigureAwait(false);
             return result;
         }
 
-        private static async Task<T> GetFirstNonNullResult<T>(List<Task<T>> tasks)
-            where T : class
+        private static async Task<string> GetFirstNonNullResult(List<Task<string?>> tasks)
```

Comment:
```suggestion
        private static async Task<string?> GetFirstNonNullResult(List<Task<string?>> tasks)
```

There are codes checking against null for the call chain:

```
FindBinaryAsync
DownloadFileWorker
SearchSymbolServerForFile
GetFirstNonNullResult
```

https://github.com/microsoft/clrmd/blob/19725d98eae81948c7c9dbbfc85cdd65217cc636/src/Microsoft.Diagnostics.Runtime/src/Utilities/Symbols/DefaultSymbolLocator.Async.cs#L54

Reply:
Thanks!

I am going to leave this for later.  Going through the symbol locator code is on my todo list so I'm hesitant to do more cleanups there until I know what the final result will look like.